cylc/flow/scripts/lint.py

```diff
@@ -663,6 +664,13 @@ def get_option_parser() -> COP:
         metavar="CODE",
         choices=tuple([f'S{i["index"]:03d}' for i in STYLE_CHECKS.values()])
     )
+    parser.add_option(
+        '--error-on-issues', '-e',
+        help='Return an error code if any issues are raised.',
+        action='store_true',
+        default=False,
```

Comment:
I think this is a sensible default:

```suggestion
        default=True,
```

Reply:
Ok, thanks for pointing me at the tests and also the suggestion about `--exit-zero`.

Reply:
Changed to `--exit-zero`, default of False is being kept because of this. Other comments have been taken into account.